datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -727,12 +735,12 @@ pub fn build_equal_condition_join_indices(
         // For every item on the build and probe we check if it matches
         // This possibly contains rows with hash collisions,
         // So we have to check here whether rows are equal or not
-        if let Some((_, indices)) = build_hashmap
-            .0
+        if let Some((_, index)) = build_hashmap
+            .map
             .get(*hash_value, |(hash, _)| *hash_value == *hash)
         {
-            for &i in indices {
-                // Check hash collisions
+            let mut i = *index - 1;
+            loop {
                 let offset_build_index = i as usize - offset_value;
```

Comment:
Offset is not necessary for usual hash join, you can remove it safely.

Reply:
👍